### PR TITLE
Adds Supplybot, a cargo-focused deliverybot with internal storage.

### PR DESCRIPTION
**Pull request by ItsMeThom**
## About the PR

Added the ~~mulebot~~ (Supplybot, see comments below) a new ghostrole/constructable lesser bot with internal storage for cargo delivery.
## Why / Balance

Cargo have a lot on their hands, so having a helpful delivery bot they can loads with goods (Storage size 250) to bring to departments will let them spend more time ~~plotting cargonian revolution~~ focused on filling out paperwork.

## Technical details

Code is heavily based on taxibot, but removes the seat and adds internal storage compartment. . I had wanted to have the supplybot able to load crates (in the same way as the ripley powerloader) but thats beyond my skill currently. perhaps later there can be an advanced mule with small hydraulic clamps that can load crates.

## Media

Pics of the supplybot:
This is the finalised mulebot (see comments below):
image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- added: Nanotrasen have delivered new designs for the Supplybot, a helpful cargo delivery bot that can ferry deliveries between departments!

### CODE REVIEW

**Comment by lzk228** [comment]
sprite is too bad

File: Resources/Prototypes/Recipes/Crafting/Graphs/bots/mulebot.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+          sprite: Objects/Misc/proximity_sensor.rsi
+          state: icon
+        name: proximity sensor
+      - tag: BorgHead
+        icon:
+          sprite: Objects/Specific/Robotics/cyborg_parts.rsi
+          state: borg_head
+        name: borg head
+        doAfter: 1
+      - tag: BorgArm
```
Comment (lzk228) [inline Resources/Prototypes/Recipes/Crafting/Graphs/bots/mulebot.yml:20]: for what? if it has no arms eventually

File: Resources/Prototypes/Entities/Mobs/NPCs/silicon.yml [hunk 2]
```diff
@@ -168,6 +168,70 @@
...
+    description: ghost-role-information-mulebot-description
+  - type: GhostTakeoverAvailable
+  - type: MovementSpeedModifier
+    weightlessModifier: 0
+    acceleration: 2
+    friction: 2
+    frictionNoInput: 6
+    baseWalkSpeed: 3
+    baseSprintSpeed: 6
+  - type: AmbientSound
```
Comment (lzk228) [inline Resources/Prototypes/Entities/Mobs/NPCs/silicon.yml:201]: i think it shouldn't always produce sounds
imagine someone who playing on mulebot and listening that engine sound all the time

**Comment by mirrorcult** [comment]
sprite looks pretty cute i think, could probably use more shading but i'll leave that up to art directors

**Comment by mirrorcult** [comment]
(or can just use MULE sprites from ss13, maybe edited to make it clear its a ghost role)

File: Resources/Prototypes/Recipes/Crafting/Graphs/bots/mulebot.yml (hunk 1 above)
Comment (ItsMeThom) [reply Resources/Prototypes/Recipes/Crafting/Graphs/bots/mulebot.yml]: Youre right, it was again, carried over from all other lesser borgs (cleanbot etc) which all require at a minimum, one borg arm. I have removed it.

**Comment by ItsMeThom** [comment]
@mirrorcult thanks, Im not a spriter so its very much programmer art as a placeholder. I tried to match it with the style of the taxibot and give it cargo colours. 

>(or can just use MULE sprites from ss13, maybe edited to make it clear its a ghost role)

Im not sure how to convert from byonds DMI format to png's for use in SS14, are there any resources I can look at?

**Comment by mirrorcult** [comment]
https://github.com/space-wizards/RSIEdit

**Comment by potato1234x** [comment]
> > sprite is too bad
> 
> Do you have any specific feedback?

just take it from tg

**Comment by ItsMeThom** [comment]
image
Fantastically resprited by @lzk228. Renamed from Mulebot to supplybot, so it won't conflict with the real M.U.L.E bot when it gets added in the future.

**Comment by ItsMeThom** [comment]
image
Latest sprits from @lzk228

**Comment by ItsMeThom** [comment]
@lzk228 are you happy enough with the sprites that we can make it resolved?

**Comment by ItsMeThom** [comment]
image
Finalised sprite for Supplybot by lzk228

**Pull request merged**